msgcntr/messageforums-component-impl/src/java/org/sakaiproject/component/app/messageforums/AnonymousManagerImpl.java

```diff
@@ -0,0 +1,198 @@
+package org.sakaiproject.component.app.messageforums;
+
+import java.sql.SQLException;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Map;
+import java.util.Random;
+import java.util.Set;
+
+import org.apache.commons.collections.CollectionUtils;
+import org.apache.commons.logging.Log;
+import org.apache.commons.logging.LogFactory;
+import org.hibernate.Hibernate;
+import org.hibernate.HibernateException;
+import org.hibernate.Query;
+import org.hibernate.Session;
+import org.hibernate.exception.ConstraintViolationException;
+import org.sakaiproject.api.app.messageforums.AnonymousManager;
+import org.sakaiproject.api.app.messageforums.AnonymousMapping;
+import org.sakaiproject.component.app.messageforums.dao.hibernate.AnonymousMappingImpl;
+import org.springframework.dao.DataIntegrityViolationException;
+import org.springframework.orm.hibernate3.HibernateCallback;
+import org.springframework.orm.hibernate3.support.HibernateDaoSupport;
+
+public class AnonymousManagerImpl extends HibernateDaoSupport implements AnonymousManager
+{
+	private static final Log LOG = LogFactory.getLog(AnonymousManagerImpl.class);
+
+	private final String ANON_ID_PADDING = "000000";
+
+	private final String QUERY_BY_SITE = "findMappingsBySite";
+	private final String QUERY_BY_SITE_AND_USERS = "findMappingsBySiteAndUsers";
+
+	private final int MAX_IN_CLAUSE_SIZE = 1000;
+
+	public String getAnonId(final String siteId, final String userId)
```

Comment:
You can use the following JavaDoc on these methods to inherit the docs from the interface:

`/**
* {@inheritDoc}
*/`